mk/compile.mk

```diff
@@ -50,9 +50,12 @@ WARNS		?= 3
 
 comp-cflags$(sm)	+= $(comp-cflags-warns-$(WARNS))
 
+CHECK ?= sparse
```

Comment:
It's called `CHECK` in the Linux kernel makefile, I guess the `C` in `C=1` comes from check also. We're printing `CHECK` when processing the C files. I think it makes more sense to keep it as it is.